### PR TITLE
Files added

### PR DESCRIPTION
**Pull request by Ayush-kumar1**
This is the backend for the eCommerce app. Pls, review it.

### CODE REVIEW

File: src/db/db.connect.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const mongoose = require("mongoose")
+
+// TODO: move to .env/sec
+// TODO: use async await instead of then/catch
+function initializeDBConnection() {
+    // Connecting to DB
```
Comment (asmitzz) [inline src/db/db.connect.js]: use try catch block here

File: src/routes/product.router.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+
+
+
+router.route("/:productId")
+    .get((req, res) => {
+        let { product } = req;
+        product.__v = undefined;
+        res.json({ success: true, product })
+
```
Comment (asmitzz) [inline src/routes/product.router.js]: remove unnecessay empty lines

**Comment by asmitzz** [comment]
overall backend code is so clean and readable. Nice work

File: src/routes/product.router.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            res.json({ success: true, product: savedProduct })
+        } catch (err) {
+            res.status(500).json({ success: false, message: "unable to add products", errorMessage: err.message })
+        }
+    })
+
+
+router.param("productId", async(req, res, next, productId) => {
+    try {
+        // Step Two: Getting the product using `findById`
```
Comment (jugalkpatel) [inline src/routes/product.router.js]: remove unnecessary comments. code itself is enough to understand the flow.

**Comment by jugalkpatel** [comment]
app became blank while removing item from wishlist page.

File: Model/Cart.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import mongoose from "mongoose";
```
Comment (thesudeshdas) [inline Model/Cart.js:1]: Change file name

File: middleware/requireLogin.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+const requireLogin = (req, res, next) => {
+
+    const { authorization } = req.headers;
+
+    if (!authorization) {
+        return res.status(401).json({ error: "You must be logged in to get access" })
+    }
+
+    const token = authorization.replace("Bearer ", "")
+    jwt.verify(token, JWT_SECRET, (err, payload) => {
```
Comment (thesudeshdas) [inline middleware/requireLogin.js:14]: Change to .env variable

**Review by thesudeshdas — APPROVED** [review]

**Pull request merged**